python/cugraph/tests/test_random_walks.py

```diff
@@ -0,0 +1,154 @@
+DATASETS = [pytest.param(d) for d in utils.DATASETS]
+DATASETS_SMALL = [pytest.param(d) for d in utils.DATASETS_SMALL]
+
+
+def calc_random_walks(
+    graph_file,
+    directed=False,
+    max_depth=None
+):
+    """
+    compute random walks for each nodes in 'start_vertices'
+
+    parameters
+    ----------
+    G : cuGraph.Graph or networkx.Graph
+        The graph can be either directed (DiGraph) or undirected (Graph).
+        Weights in the graph are ignored.
+        Use weight parameter if weights need to be considered
+        (currently not supported)
+
+    start_vertices : int or list or cudf.Series
+        A single node or a list or a cudf.Series of nodes from which to run
+        the random walks
+
+    max_depth : int
+        The maximum depth of the random walks
+
+
+    Returns
+    -------
+    random_walks_edge_lists : cudf.DataFrame
+        GPU data frame containing all random walks sources identifiers,
+        destination identifiers, edge weights
+
+    seeds_offsets: cudf.Series
+        Series containing the starting offset in the returned edge list
+        for each vertex in start_vertices.
+    """
+    G = utils.generate_cugraph_graph_from_file(
+        graph_file, directed=directed, edgevals=True)
+    assert G is not None
+
+    k = random.randint(1, 10)
+    start_vertices = random.sample(range(G.number_of_vertices()), k)
+    df, offsets = cugraph.random_walks(G, start_vertices, max_depth)
+
+    return df, offsets, start_vertices
+
+
+def check_random_walks(df, offsets, seeds, df_G=None):
+    invalid_edge = 0
+    invalid_seeds = 0
+    invalid_weight = 0
+    offsets_idx = 0
+    for i in range(len(df.index)):
+        src, dst, weight = df.iloc[i].to_array()
+        if i == offsets[offsets_idx]:
+            if df['src'].iloc[i] != seeds[offsets_idx]:
+                invalid_seeds += 1
+                print(
+                        "[ERR] Invalid seed: "
+                        " src {} != src {}"
+                        .format(df['src'].iloc[i], offsets[offsets_idx])
+                    )
+            offsets_idx += 1
+
+        edge = df.loc[(df['src'] == (src)) & (df['dst'] == (dst))].reset_index(
+            drop=True)
+        exp_edge = df_G.loc[
+            (df_G['src'] == (src)) & (
+                df_G['dst'] == (dst))].reset_index(drop=True)
+
+        if not exp_edge.equals(edge[:1]):
+            print(
+                    "[ERR] Invalid edge: "
+                    "There is no edge src {} dst {} weight {}"
+                    .format(src, dst, weight)
+                )
+            invalid_weight += 1
+
+    assert invalid_edge == 0
+    assert invalid_seeds == 0
+    assert invalid_weight == 0
+
+# =============================================================================
+# Pytest Setup / Teardown - called for each test function
+# =============================================================================
+
+
+def prepare_test():
```

Comment:
If you make this a setup function, pytest will automatically call it for you before each test, as done [here](https://github.com/rapidsai/cugraph/blob/79c3ba059d2cc4f2bfafece0e92671013a584175/python/cugraph/tests/test_bfs.py#L75).